diff --git a/core/modules/system/tests/src/Functional/Database/TemporaryQueryTest.php b/core/modules/system/tests/src/Functional/Database/TemporaryQueryTest.php
index 5226585..1731510 100644
--- a/core/modules/system/tests/src/Functional/Database/TemporaryQueryTest.php
+++ b/core/modules/system/tests/src/Functional/Database/TemporaryQueryTest.php
@@ -26,6 +26,7 @@ public function countTableRows($table_name) {
    */
   public function testTemporaryQuery() {
     $this->drupalGet('database_test/db_query_temporary');
+throw new \Exception($this->getSession()->getPage()->getContent());    
     $data = json_decode($this->getSession()->getPage()->getContent());
     if ($data) {
       $this->assertEqual($this->countTableRows('test'), $data->row_count, 'The temporary table contains the correct amount of rows.');
