diff --git a/core/tests/Drupal/KernelTests/KernelTestBase.php b/core/tests/Drupal/KernelTests/KernelTestBase.php
index 6a35f15..1d14f62 100644
--- a/core/tests/Drupal/KernelTests/KernelTestBase.php
+++ b/core/tests/Drupal/KernelTests/KernelTestBase.php
@@ -2,6 +2,7 @@
 
 namespace Drupal\KernelTests;
 
+use Drupal\Component\Utility\Timer;
 use Drupal\Component\FileCache\ApcuFileCacheBackend;
 use Drupal\Component\FileCache\FileCache;
 use Drupal\Component\FileCache\FileCacheFactory;
@@ -225,6 +226,7 @@ protected static function getDrupalRoot() {
    * {@inheritdoc}
    */
   protected function setUp() {
+Timer::start('drudbal:install_cli');
     parent::setUp();
 
     $this->root = static::getDrupalRoot();
@@ -621,11 +623,13 @@ protected function tearDown() {
     $test_connection_info = Database::getConnectionInfo('default');
     $test_prefix = $test_connection_info['default']['prefix']['default'];
     if ($original_prefix != $test_prefix) {
-      $tables = Database::getConnection()->schema()->findTables('%');
+      //$tables = Database::getConnection()->schema()->findTables('%');
+      $tables = Database::getConnection()->getDbalConnection()->getSchemaManager()->listTableNames();
       foreach ($tables as $table) {
-        if (Database::getConnection()->schema()->dropTable($table)) {
+        //if (Database::getConnection()->schema()->dropTable($table)) {
+          Database::getConnection()->getDbalConnection()->getSchemaManager()->dropTable($table);
           unset($tables[$table]);
-        }
+        //}
       }
     }
 
