diff --git a/core/tests/Drupal/KernelTests/KernelTestBase.php b/core/tests/Drupal/KernelTests/KernelTestBase.php
index 6a35f15..e9599a6 100644
--- a/core/tests/Drupal/KernelTests/KernelTestBase.php
+++ b/core/tests/Drupal/KernelTests/KernelTestBase.php
@@ -2,6 +2,7 @@
 
 namespace Drupal\KernelTests;
 
+use Drupal\Component\Utility\Timer;
 use Drupal\Component\FileCache\ApcuFileCacheBackend;
 use Drupal\Component\FileCache\FileCache;
 use Drupal\Component\FileCache\FileCacheFactory;
@@ -225,12 +226,18 @@ protected static function getDrupalRoot() {
    * {@inheritdoc}
    */
   protected function setUp() {
+Timer::start('drudbal:install_cli');
     parent::setUp();
+error_log(Timer::read('drudbal:install_cli') . '||TestCase::setup done');
 
     $this->root = static::getDrupalRoot();
+error_log(Timer::read('drudbal:install_cli') . '||getDrupalRoot() done');
     $this->initFileCache();
+error_log(Timer::read('drudbal:install_cli') . '||initFileCache() done');
     $this->bootEnvironment();
+error_log(Timer::read('drudbal:install_cli') . '||bootEnvironment() done');
     $this->bootKernel();
+error_log(Timer::read('drudbal:install_cli') . '||bootKernel() done');
   }
 
   /**
@@ -615,6 +622,7 @@ protected function tearDown() {
       $this->kernel->shutdown();
     }
 
+error_log(Timer::read('drudbal:install_cli') . '||Before removing tables');
     // Remove all prefixed tables.
     $original_connection_info = Database::getConnectionInfo('simpletest_original_default');
     $original_prefix = $original_connection_info['default']['prefix']['default'];
@@ -660,6 +668,7 @@ protected function tearDown() {
     new Settings([]);
 
     parent::tearDown();
+error_log(Timer::read('drudbal:install_cli') . '||Final');
   }
 
   /**
