diff --git a/core/tests/Drupal/Tests/UpdatePathTestTrait.php b/core/tests/Drupal/Tests/UpdatePathTestTrait.php
index 61facab0bd..e17b4e2f0f 100644
--- a/core/tests/Drupal/Tests/UpdatePathTestTrait.php
+++ b/core/tests/Drupal/Tests/UpdatePathTestTrait.php
@@ -2,6 +2,7 @@
 
 namespace Drupal\Tests;
 
+use Drupal\Core\Database\Database;
 use Drupal\Core\Url;
 
 /**
@@ -142,6 +143,8 @@ protected function runUpdates($update_url = NULL) {
         $this->assertFalse($needs_updates, 'After all updates ran, entity schema is up to date.');
       }
     }
+
+    Database::getConnection()->schema()->dbalSchemaForceReload();
   }
 
   /**
