diff --git a/core/modules/system/tests/src/Functional/Entity/Update/LangcodeToAsciiUpdateTest.php b/core/modules/system/tests/src/Functional/Entity/Update/LangcodeToAsciiUpdateTest.php
index d1a3cab..3720a8b 100644
--- a/core/modules/system/tests/src/Functional/Entity/Update/LangcodeToAsciiUpdateTest.php
+++ b/core/modules/system/tests/src/Functional/Entity/Update/LangcodeToAsciiUpdateTest.php
@@ -27,7 +27,7 @@ public function setDatabaseDumpFiles() {
   public function testLangcodeColumnCollation() {
     // Only testable on MySQL.
     // @see https://www.drupal.org/node/301038
-    if (Database::getConnection()->databaseType() !== 'mysql') {
+    if (Database::getConnection()->driver() !== 'mysql') {
       $this->pass('This test can only run on MySQL');
       return;
     }
