diff --git a/core/modules/field/tests/modules/field_test/field_test.field.inc b/core/modules/field/tests/modules/field_test/field_test.field.inc
index 5edd25d..65f7627 100644
--- a/core/modules/field/tests/modules/field_test/field_test.field.inc
+++ b/core/modules/field/tests/modules/field_test/field_test.field.inc
@@ -41,6 +41,7 @@ function field_test_default_value(FieldableEntityInterface $entity, FieldDefinit
  * Implements hook_entity_field_access().
  */
 function field_test_entity_field_access($operation, FieldDefinitionInterface $field_definition, AccountInterface $account, FieldItemListInterface $items = NULL) {
+throw new \Exception('pippo');
   if ($field_definition->getName() == "field_no_{$operation}_access") {
     return AccessResult::forbidden();
   }
diff --git a/core/modules/system/tests/src/Functional/Entity/EntityRevisionsTest.php b/core/modules/system/tests/src/Functional/Entity/EntityRevisionsTest.php
index caf5fa2..0026749 100644
--- a/core/modules/system/tests/src/Functional/Entity/EntityRevisionsTest.php
+++ b/core/modules/system/tests/src/Functional/Entity/EntityRevisionsTest.php
@@ -117,6 +117,8 @@ protected function runRevisionsTests($entity_type) {
     $entity = $this->container->get('entity_type.manager')
       ->getStorage($entity_type)
       ->load($entity->id->value);
+\Drupal::database()->getDbalExtension()->setDebugging(TRUE);
+error_log($entity_type . '/manage/' . $entity->id->value . '/edit');
     $this->drupalGet($entity_type . '/manage/' . $entity->id->value . '/edit');
     $this->assertFieldById('edit-name-0-value', $entity->name->value, format_string('%entity_type: Name matches in UI.', ['%entity_type' => $entity_type]));
     $this->assertFieldById('edit-field-test-text-0-value', $entity->field_test_text->value, format_string('%entity_type: Text matches in UI.', ['%entity_type' => $entity_type]));
diff --git a/core/modules/system/tests/src/Functional/Entity/EntityViewControllerTest.php b/core/modules/system/tests/src/Functional/Entity/EntityViewControllerTest.php
index 684e0be..aa995d0 100644
--- a/core/modules/system/tests/src/Functional/Entity/EntityViewControllerTest.php
+++ b/core/modules/system/tests/src/Functional/Entity/EntityViewControllerTest.php
@@ -97,6 +97,8 @@ public function testFieldItemAttributes() {
 
     // Browse to the entity and verify that the attribute is rendered in the
     // field item HTML markup.
+\Drupal::database()->getDbalExtension()->setDebugging(TRUE);
+error_log($entity->id());
     $this->drupalGet('entity_test/' . $entity->id());
     $xpath = $this->xpath('//div[@data-field-item-attr="foobar"]/p[text()=:value]', [':value' => $test_value]);
     $this->assertTrue($xpath, 'The field item attribute has been found in the rendered output of the field.');
