diff --git a/core/modules/field/tests/modules/field_test/field_test.field.inc b/core/modules/field/tests/modules/field_test/field_test.field.inc
index 5edd25d..a81e761 100644
--- a/core/modules/field/tests/modules/field_test/field_test.field.inc
+++ b/core/modules/field/tests/modules/field_test/field_test.field.inc
@@ -48,6 +48,7 @@ function field_test_entity_field_access($operation, FieldDefinitionInterface $fi
   // Only grant view access to test_view_field fields when the user has
   // 'view test_view_field content' permission.
   if ($field_definition->getName() == 'test_view_field' && $operation == 'view') {
+throw new \Exception($account->hasPermission('view test_view_field content') ? 'Auth' : 'Not Auth');
     return AccessResult::forbiddenIf(!$account->hasPermission('view test_view_field content'))->cachePerPermissions();
   }
 
diff --git a/core/modules/system/tests/src/Functional/Entity/EntityRevisionsTest.php b/core/modules/system/tests/src/Functional/Entity/EntityRevisionsTest.php
index caf5fa2..0026749 100644
--- a/core/modules/system/tests/src/Functional/Entity/EntityRevisionsTest.php
+++ b/core/modules/system/tests/src/Functional/Entity/EntityRevisionsTest.php
@@ -117,6 +117,8 @@ protected function runRevisionsTests($entity_type) {
     $entity = $this->container->get('entity_type.manager')
       ->getStorage($entity_type)
       ->load($entity->id->value);
+\Drupal::database()->getDbalExtension()->setDebugging(TRUE);
+error_log($entity_type . '/manage/' . $entity->id->value . '/edit');
     $this->drupalGet($entity_type . '/manage/' . $entity->id->value . '/edit');
     $this->assertFieldById('edit-name-0-value', $entity->name->value, format_string('%entity_type: Name matches in UI.', ['%entity_type' => $entity_type]));
     $this->assertFieldById('edit-field-test-text-0-value', $entity->field_test_text->value, format_string('%entity_type: Text matches in UI.', ['%entity_type' => $entity_type]));
diff --git a/core/modules/system/tests/src/Functional/Entity/EntityViewControllerTest.php b/core/modules/system/tests/src/Functional/Entity/EntityViewControllerTest.php
index 684e0be..aa995d0 100644
--- a/core/modules/system/tests/src/Functional/Entity/EntityViewControllerTest.php
+++ b/core/modules/system/tests/src/Functional/Entity/EntityViewControllerTest.php
@@ -97,6 +97,8 @@ public function testFieldItemAttributes() {
 
     // Browse to the entity and verify that the attribute is rendered in the
     // field item HTML markup.
+\Drupal::database()->getDbalExtension()->setDebugging(TRUE);
+error_log($entity->id());
     $this->drupalGet('entity_test/' . $entity->id());
     $xpath = $this->xpath('//div[@data-field-item-attr="foobar"]/p[text()=:value]', [':value' => $test_value]);
     $this->assertTrue($xpath, 'The field item attribute has been found in the rendered output of the field.');
