diff --git a/lib/Doctrine/DBAL/Driver/PDOSqlite/Driver.php b/lib/Doctrine/DBAL/Driver/PDOSqlite/Driver.php
index 4fe4e72..2be3617 100644
--- a/lib/Doctrine/DBAL/Driver/PDOSqlite/Driver.php
+++ b/lib/Doctrine/DBAL/Driver/PDOSqlite/Driver.php
@@ -59,6 +59,9 @@ class Driver extends AbstractSQLiteDriver
                 $driverOptions
             );
         } catch (PDOException $ex) {
+if ($dsn !== 'sqlite:sites/default/files/.ht.sqlite') {
+            error_log('herehere: '. var_export([$this->_constructPdoDsn($params),$username,$password,$driverOptions], TRUE) . ' :herehere');
+}
             throw DBALException::driverException($this, $ex);
         }
 
