diff --git a/lib/Doctrine/DBAL/Platforms/OraclePlatform.php b/lib/Doctrine/DBAL/Platforms/OraclePlatform.php
index dc0217d..de2b747 100644
--- a/lib/Doctrine/DBAL/Platforms/OraclePlatform.php
+++ b/lib/Doctrine/DBAL/Platforms/OraclePlatform.php
@@ -411,6 +411,33 @@ class OraclePlatform extends AbstractPlatform
     }
 
     /**
+     * Returns the list of all the indexes in the database.
+     *
+     * @param string $database
+     *
+     * @return string
+     */
+    public function getListAllIndexesSQL(string $database): string
+    {
+        $databaseIdentifier = $this->normalizeIdentifier($database);
+        $quotedDatabaseIdentifier = $this->quoteStringLiteral($databaseIdentifier->getName());
+        return <<<SQL
+          SELECT ind_col.table_name as table_name,
+                 ind_col.index_name AS name,
+                 ind.index_type AS type,
+                 decode(ind.uniqueness, 'NONUNIQUE', 0, 'UNIQUE', 1) AS is_unique,
+                 ind_col.column_name AS column_name,
+                 ind_col.column_position AS column_pos,
+                 con.constraint_type AS is_primary
+            FROM all_ind_columns ind_col
+       LEFT JOIN all_indexes ind ON ind.owner = ind_col.index_owner AND ind.index_name = ind_col.index_name
+       LEFT JOIN all_constraints con ON  con.owner = ind_col.index_owner AND con.index_name = ind_col.index_name
+           WHERE ind_col.index_owner = $quotedDatabaseIdentifier
+        ORDER BY ind_col.table_name, ind_col.index_name, ind_col.column_position
+SQL;
+    }
+
+    /**
      * {@inheritDoc}
      *
      * @license New BSD License
@@ -606,6 +633,33 @@ END;';
     }
 
     /**
+     * Returns the list of all the foreign keys in the database.
+     *
+     * @param string $database
+     *
+     * @return string
+     */
+    public function getListAllForeignKeysSQL(string $database): string
+    {
+        $databaseIdentifier = $this->normalizeIdentifier($database);
+        $quotedDatabaseIdentifier = $this->quoteStringLiteral($databaseIdentifier->getName());
+        return <<<SQL
+          SELECT cols.table_name,
+                 alc.constraint_name,
+                 alc.DELETE_RULE,
+                 cols.column_name "local_column",
+                 cols.position,
+                 r_cols.table_name "references_table",
+                 r_cols.column_name "foreign_column"
+            FROM all_cons_columns cols
+       LEFT JOIN all_constraints alc ON alc.owner = cols.owner AND alc.constraint_name = cols.constraint_name
+       LEFT JOIN all_cons_columns r_cols ON r_cols.owner = alc.r_owner AND r_cols.constraint_name = alc.r_constraint_name AND r_cols.position = cols.position
+           WHERE cols.owner = $quotedDatabaseIdentifier AND alc.constraint_type = 'R'
+        ORDER BY cols.table_name, cols.constraint_name, cols.position
+SQL;
+    }
+
+    /**
      * {@inheritDoc}
      */
     public function getListTableForeignKeysSQL($table)
@@ -649,6 +703,27 @@ END;';
     }
 
     /**
+     * Returns the list of all the columns of all the tables in the database.
+     *
+     * @param string $database
+     *
+     * @return string
+     */
+    public function getListAllColumnsSQL(string $database): string
+    {
+        $databaseIdentifier = $this->normalizeIdentifier($database);
+        $quotedDatabaseIdentifier = $this->quoteStringLiteral($databaseIdentifier->getName());
+        return <<<SQL
+          SELECT c.*,
+                 d.comments AS comments
+            FROM all_tab_columns c
+       LEFT JOIN all_col_comments d ON d.OWNER = c.OWNER AND d.TABLE_NAME = c.TABLE_NAME AND d.COLUMN_NAME = c.COLUMN_NAME
+           WHERE c.owner = $quotedDatabaseIdentifier
+        ORDER BY c.table_name, c.column_id
+SQL;
+    }
+
+    /**
      * {@inheritDoc}
      */
     public function getListTableColumnsSQL($table, $database = null)
diff --git a/lib/Doctrine/DBAL/Schema/OracleSchemaManager.php b/lib/Doctrine/DBAL/Schema/OracleSchemaManager.php
index 9feecdc..d6be078 100644
--- a/lib/Doctrine/DBAL/Schema/OracleSchemaManager.php
+++ b/lib/Doctrine/DBAL/Schema/OracleSchemaManager.php
@@ -36,6 +36,65 @@ class OracleSchemaManager extends AbstractSchemaManager
     /**
      * {@inheritdoc}
      */
+    public function listTables()
+    {
+        $currentDatabase = $this->_conn->getDatabase();
+
+        $tableNames = $this->listTableNames();
+
+        // Get all column definitions in one database call.
+        $columnsByTable = $this->getAssetRecordsByTable($this->_platform->getListAllColumnsSQL($currentDatabase));
+
+        // Get all foreign keys definitions in one database call.
+        $foreignKeysByTable = $this->getAssetRecordsByTable($this->_platform->getListAllForeignKeysSQL($currentDatabase));
+
+        // Get all indexes definitions in one database call.
+        $indexesByTable = $this->getAssetRecordsByTable($this->_platform->getListAllIndexesSQL($currentDatabase));
+
+        $tables = [];
+        foreach ($tableNames as $tableName) {
+            $unquotedTableName = trim($tableName, '"');
+
+            $columns = $this->_getPortableTableColumnList($tableName, null, $columnsByTable[$unquotedTableName]);
+
+            $foreignKeys = [];
+            if (isset($foreignKeysByTable[$unquotedTableName])) {
+                $foreignKeys = $this->_getPortableTableForeignKeysList($foreignKeysByTable[$unquotedTableName]);
+            }
+
+            $indexes = [];
+            if (isset($indexesByTable[$unquotedTableName])) {
+                $indexes = $this->_getPortableTableIndexesList($indexesByTable[$unquotedTableName], $tableName);
+            }
+
+            $tables[] = new Table($tableName, $columns, $indexes, $foreignKeys, false, []);
+        }
+
+        return $tables;
+    }
+
+    /**
+     * Helper method to group a set of asset records by the table name.
+     *
+     * @param string $sql An SQL statement to be executed, that contains a
+     *                    TABLE_NAME field for grouping.
+     *
+     * @return array An associative array with key being the table name, and
+     *               value a simple array of records associated with the table.
+     */
+    private function getAssetRecordsByTable(string $sql): array
+    {
+        $input = $this->_conn->fetchAll($sql);
+        $output = [];
+        foreach ($input as $record) {
+            $output[$record['TABLE_NAME']][] = $record;
+        }
+        return $output;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
     public function dropDatabase($database)
     {
         try {
diff --git a/tests/Doctrine/Tests/DBAL/Functional/Schema/OracleSchemaManagerTest.php b/tests/Doctrine/Tests/DBAL/Functional/Schema/OracleSchemaManagerTest.php
index bea2691..00f5a64 100644
--- a/tests/Doctrine/Tests/DBAL/Functional/Schema/OracleSchemaManagerTest.php
+++ b/tests/Doctrine/Tests/DBAL/Functional/Schema/OracleSchemaManagerTest.php
@@ -273,4 +273,29 @@ class OracleSchemaManagerTest extends SchemaManagerFunctionalTestCase
         $this->assertSame('datetime', $columns['col_datetime']->getType()->getName());
         $this->assertSame('datetimetz', $columns['col_datetimetz']->getType()->getName());
     }
+
+    /**
+     * @group DBAL-2766
+     */
+    public function testCreateSchemaNumberOfQueriesInvariable(): void
+    {
+        // Introspect the db schema.
+        $preCount = $this->_sqlLoggerStack->currentQuery;
+        $schema = $this->_sm->createSchema();
+        $firstQueryCount = $this->_sqlLoggerStack->currentQuery - $preCount;
+
+        // Create a couple of additional tables.
+        $this->_conn->executeUpdate("CREATE TABLE tbl_test_2766_0 (x_id VARCHAR2(255) DEFAULT 'x' NOT NULL, x_data CLOB DEFAULT NULL NULL, x_number NUMBER(10) DEFAULT 0 NOT NULL, PRIMARY KEY(x_id))");
+        $this->_conn->executeUpdate("CREATE TABLE tbl_test_2766_1 (x_id VARCHAR2(255) DEFAULT 'x' NOT NULL, x_data CLOB DEFAULT NULL NULL, x_number NUMBER(10) DEFAULT 0 NOT NULL, x_parent_id VARCHAR2(255) DEFAULT 'x' NOT NULL, CONSTRAINT tbl_test_2766_fk_1 FOREIGN KEY (x_parent_id) REFERENCES tbl_test_2766_0(x_id), PRIMARY KEY(x_id))");
+        $this->_conn->executeUpdate("CREATE UNIQUE INDEX tbl_test_2766_uix_1 ON tbl_test_2766_1 (x_number)");
+
+        // Introspect the db schema again.
+        $preCount = $this->_sqlLoggerStack->currentQuery;
+        $schema = $this->_sm->createSchema();
+        $secondQueryCount = $this->_sqlLoggerStack->currentQuery - $preCount;
+
+        // The number of queries needed to execute createSchema should be the
+        // same regardless of additional tables added.
+        $this->assertEquals($firstQueryCount, $secondQueryCount);
+    }
 }
