diff --git a/core/lib/Drupal/Core/Test/TestDiscovery.php b/core/lib/Drupal/Core/Test/TestDiscovery.php
index 1c319dfc..8b4f1325 100644
--- a/core/lib/Drupal/Core/Test/TestDiscovery.php
+++ b/core/lib/Drupal/Core/Test/TestDiscovery.php
@@ -270,6 +270,18 @@ public function findAllClassFiles($extension = NULL) {
    * @see https://www.drupal.org/node/2296635
    */
   public static function scanDirectory($namespace_prefix, $path) {
+    static $skipped_classes;
+
+    if (!isset($skipped_classes)) {
+      $env = getenv();
+      if (isset($env['PHPUNIT_SKIP_CLASS'])) {
+        $skipped_classes = json_decode($env['PHPUNIT_SKIP_CLASS']);
+      }
+      else {
+        $skipped_classes = [];
+      }
+    }
+
     if (substr($namespace_prefix, -1) !== '\\') {
       throw new \InvalidArgumentException("Namespace prefix for $path must contain a trailing namespace separator.");
     }
@@ -300,6 +312,9 @@ public static function scanDirectory($namespace_prefix, $path) {
         $class .= strtr($subpath, '/', '\\') . '\\';
       }
       $class .= $fileinfo->getBasename('.php');
+      if (in_array($class, $skipped_classes)) {
+        continue;
+      }
       $classes[$class] = $fileinfo->getPathname();
     }
     return $classes;
