diff --git a/core/lib/Drupal/Core/Mail/MailManager.php b/core/lib/Drupal/Core/Mail/MailManager.php
index 05c0c8adfa..c76bad110c 100644
--- a/core/lib/Drupal/Core/Mail/MailManager.php
+++ b/core/lib/Drupal/Core/Mail/MailManager.php
@@ -285,6 +285,7 @@ public function doMail($module, $key, $to, $langcode, $params = [], $reply = NUL
     // Attempt to convert relative URLs to absolute.
     foreach ($message['body'] as &$body_part) {
       if ($body_part instanceof MarkupInterface) {
+dump([$body_part, \Drupal::request()->getSchemeAndHttpHost()]);
         $body_part = Markup::create(Html::transformRootRelativeUrlsToAbsolute((string) $body_part, \Drupal::request()->getSchemeAndHttpHost()));
       }
     }
