diff --git a/core/tests/Drupal/Tests/UpdatePathTestTrait.php b/core/tests/Drupal/Tests/UpdatePathTestTrait.php
index 0abf1cc477..992dc38ec9 100644
--- a/core/tests/Drupal/Tests/UpdatePathTestTrait.php
+++ b/core/tests/Drupal/Tests/UpdatePathTestTrait.php
@@ -44,6 +44,7 @@ protected function runUpdates($update_url = NULL) {
     ]);

     $this->drupalGet($update_url);
+dump($update_url, $this->getSession()->getPage());
     $this->updateRequirementsProblem();
     $this->clickLink('Continue');

