diff --git a/core/includes/install.core.inc b/core/includes/install.core.inc
index 5a5f92339b..385fbca06b 100644
--- a/core/includes/install.core.inc
+++ b/core/includes/install.core.inc
@@ -961,6 +961,7 @@ function install_get_form($form_id, array &$install_state) {
     // values taken from the installation state.
     $install_form_id = $form_builder->getFormId($form_id, $form_state);
     if (!empty($install_state['forms'][$install_form_id])) {
+dump([$install_form_id, $install_state['forms'][$install_form_id]]);
       $form_state->setValues($install_state['forms'][$install_form_id]);
     }
     $form_builder->submitForm($form_id, $form_state);
