diff --git a/lib/Doctrine/DBAL/Schema/SqliteSchemaManager.php b/lib/Doctrine/DBAL/Schema/SqliteSchemaManager.php
index ac758015..c8986b46 100644
--- a/lib/Doctrine/DBAL/Schema/SqliteSchemaManager.php
+++ b/lib/Doctrine/DBAL/Schema/SqliteSchemaManager.php
@@ -304,6 +304,8 @@ class SqliteSchemaManager extends AbstractSchemaManager
      */
     protected function _getPortableTableColumnDefinition($tableColumn)
     {
+global $xxx;
+if ($xxx) dump(['IN', $tableColumn]);
         $parts               = explode('(', $tableColumn['type']);
         $tableColumn['type'] = trim($parts[0]);
         if (isset($parts[1])) {
@@ -373,6 +375,7 @@ class SqliteSchemaManager extends AbstractSchemaManager
             'autoincrement' => false,
         ];

+if ($xxx) dump(['OUT', $options, new Column($tableColumn['name'], Type::getType($type), $options)]);
         return new Column($tableColumn['name'], Type::getType($type), $options);
     }
