diff --git a/lib/Doctrine/DBAL/Schema/SqliteSchemaManager.php b/lib/Doctrine/DBAL/Schema/SqliteSchemaManager.php
index 0c2f4be..b94e769 100644
--- a/lib/Doctrine/DBAL/Schema/SqliteSchemaManager.php
+++ b/lib/Doctrine/DBAL/Schema/SqliteSchemaManager.php
@@ -301,6 +301,8 @@ class SqliteSchemaManager extends AbstractSchemaManager
      */
     protected function _getPortableTableColumnDefinition(array $tableColumn) : Column
     {
+global $xxx;
+if ($xxx) dump(['IN', $tableColumn]);
         preg_match('/^([^()]*)\\s*(\\(((\\d+)(,\\s*(\\d+))?)\\))?/', $tableColumn['type'], $matches);

         $dbType = trim(strtolower($matches[1]));
@@ -354,7 +356,7 @@ class SqliteSchemaManager extends AbstractSchemaManager
             'scale'     => $scale,
             'autoincrement' => false,
         ];
-
+if ($xxx) dump(['OUT', $options, new Column($tableColumn['name'], Type::getType($type), $options)]);
         return new Column($tableColumn['name'], Type::getType($type), $options);
     }
