diff --git a/lib/Doctrine/DBAL/Schema/AbstractSchemaManager.php b/lib/Doctrine/DBAL/Schema/AbstractSchemaManager.php
index 8e4ffa5f..57fadf90 100644
--- a/lib/Doctrine/DBAL/Schema/AbstractSchemaManager.php
+++ b/lib/Doctrine/DBAL/Schema/AbstractSchemaManager.php
@@ -163,6 +163,8 @@ abstract class AbstractSchemaManager
         $sql = $this->_platform->getListTableColumnsSQL($table, $database);

         $tableColumns = $this->_conn->fetchAll($sql);
+global $xxx;
+if ($xxx) dump(['COLS', $tableColumns]);

         return $this->_getPortableTableColumnList($table, $database, $tableColumns);
     }
