diff --git a/lib/Doctrine/DBAL/DriverManager.php b/lib/Doctrine/DBAL/DriverManager.php
index e3488df..c02484f 100644
--- a/lib/Doctrine/DBAL/DriverManager.php
+++ b/lib/Doctrine/DBAL/DriverManager.php
@@ -169,6 +169,7 @@ final class DriverManager
             }
         }
 
+error_log(var_export([$params], true));
         return new $wrapperClass($params, $driver, $config, $eventManager);
     }
 
