diff --git a/core/tests/Drupal/FunctionalTests/Update/UpdatePathTestBase.php b/core/tests/Drupal/FunctionalTests/Update/UpdatePathTestBase.php
index 744f527..7ccbe2f 100644
--- a/core/tests/Drupal/FunctionalTests/Update/UpdatePathTestBase.php
+++ b/core/tests/Drupal/FunctionalTests/Update/UpdatePathTestBase.php
@@ -386,6 +386,8 @@ protected function runUpdates() {
         $this->assertFalse($needs_updates, 'After all updates ran, entity schema is up to date.');
       }
     }
+
+    Database::getConnection()->schema()->dbalSchemaForceReload();
   }

   /**
