diff --git a/core/tests/Drupal/KernelTests/Core/Entity/EntityBundleFieldTest.php b/core/tests/Drupal/KernelTests/Core/Entity/EntityBundleFieldTest.php
index 22476d7..0ab572c 100644
--- a/core/tests/Drupal/KernelTests/Core/Entity/EntityBundleFieldTest.php
+++ b/core/tests/Drupal/KernelTests/Core/Entity/EntityBundleFieldTest.php
@@ -44,6 +44,7 @@ protected function setUp() {
    * Tests making use of a custom bundle field.
    */
   public function testCustomBundleFieldUsage() {
+\Drupal::database()->getDbalExtension()->setDebugging(TRUE);
     entity_test_create_bundle('custom');
 
     // Check that an entity with bundle entity_test does not have the custom
