diff --git a/core/modules/field/tests/src/Kernel/Migrate/d7/MigrateFieldTest.php b/core/modules/field/tests/src/Kernel/Migrate/d7/MigrateFieldTest.php
index 6531fe4..647e113 100644
--- a/core/modules/field/tests/src/Kernel/Migrate/d7/MigrateFieldTest.php
+++ b/core/modules/field/tests/src/Kernel/Migrate/d7/MigrateFieldTest.php
@@ -35,6 +35,7 @@ class MigrateFieldTest extends MigrateDrupal7TestBase {
    * {@inheritdoc}
    */
   protected function setUp() {
+\Drupal::database()->getDbalExtension()->setDebugging(TRUE);
     parent::setUp();
     $this->installConfig(static::$modules);
     $this->executeMigration('d7_field');
