diff --git a/core/tests/Drupal/Tests/UpdatePathTestTrait.php b/core/tests/Drupal/Tests/UpdatePathTestTrait.php
index 61facab0bd..5854c13c02 100644
--- a/core/tests/Drupal/Tests/UpdatePathTestTrait.php
+++ b/core/tests/Drupal/Tests/UpdatePathTestTrait.php
@@ -142,6 +142,8 @@ protected function runUpdates($update_url = NULL) {
         $this->assertFalse($needs_updates, 'After all updates ran, entity schema is up to date.');
       }
     }
+
+    Database::getConnection()->schema()->dbalSchemaForceReload();
   }
 
   /**
