diff --git a/core/lib/Drupal/Core/Entity/Sql/SqlContentEntityStorageSchema.php b/core/lib/Drupal/Core/Entity/Sql/SqlContentEntityStorageSchema.php
index 7f9fe4d..aeb5fae 100644
--- a/core/lib/Drupal/Core/Entity/Sql/SqlContentEntityStorageSchema.php
+++ b/core/lib/Drupal/Core/Entity/Sql/SqlContentEntityStorageSchema.php
@@ -1397,12 +1397,12 @@ protected function deleteDedicatedTableSchema(FieldStorageDefinitionInterface $s
     $table_mapping = $this->storage->getTableMapping();
     $table_name = $table_mapping->getDedicatedDataTableName($storage_definition, $storage_definition->isDeleted());
     if ($this->database->schema()->tableExists($table_name)) {
-      $this->database->schema()->dropTable($table_name);
+//      $this->database->schema()->dropTable($table_name);
     }
     if ($this->entityType->isRevisionable()) {
       $revision_table_name = $table_mapping->getDedicatedRevisionTableName($storage_definition, $storage_definition->isDeleted());
       if ($this->database->schema()->tableExists($revision_table_name)) {
-        $this->database->schema()->dropTable($revision_table_name);
+//        $this->database->schema()->dropTable($revision_table_name);
       }
     }
     $this->deleteFieldSchemaData($storage_definition);
diff --git a/core/tests/Drupal/KernelTests/Core/Entity/EntityBundleFieldTest.php b/core/tests/Drupal/KernelTests/Core/Entity/EntityBundleFieldTest.php
index 22476d7..6e3e18b 100644
--- a/core/tests/Drupal/KernelTests/Core/Entity/EntityBundleFieldTest.php
+++ b/core/tests/Drupal/KernelTests/Core/Entity/EntityBundleFieldTest.php
@@ -107,6 +107,7 @@ public function testCustomBundleFieldUsage() {
 
     // Purge field data, and check that the storage definition has been
     // completely removed once the data is purged.
+\Drupal::database()->getDbalExtension()->setDebugging(TRUE);
     field_purge_batch(10);
     $this->assertFalse($this->database->schema()->tableExists($table), 'Custom field table was deleted');
   }
