diff --git a/core/tests/Drupal/KernelTests/Core/Database/SchemaTest.php b/core/tests/Drupal/KernelTests/Core/Database/SchemaTest.php
index 2a86e3470f..859e828823 100644
--- a/core/tests/Drupal/KernelTests/Core/Database/SchemaTest.php
+++ b/core/tests/Drupal/KernelTests/Core/Database/SchemaTest.php
@@ -153,8 +153,12 @@ public function testSchema() {
 
     // Recreate the table.
     $this->schema->createTable('test_table', $table_specification);
+dump("xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx");
+dump($this->schema->dbalSchema()->getTable($this->schema->tableName('test_table')));
     $this->schema->changeField('test_table', 'test_field', 'test_field', ['type' => 'int', 'not null' => TRUE, 'default' => 0]);
+dump($this->schema->dbalSchema()->getTable($this->schema->tableName('test_table')));
     $this->schema->addField('test_table', 'test_serial', ['type' => 'int', 'not null' => TRUE, 'default' => 0, 'description' => 'Added column description.']);
+dump($this->schema->dbalSchema()->getTable($this->schema->tableName('test_table')));
 
     // Assert that the column comment has been set.
     $this->checkSchemaComment('Added column description.', 'test_table', 'test_serial');
